src/agent/agent.c

```diff
@@ -118,15 +182,18 @@ Agent *agent_new(void) {
                 return NULL;
         }
 
-        Agent *n = malloc0(sizeof(Agent));
-        n->ref_count = 1;
-        n->event = steal_pointer(&event);
-        n->user_bus_service_name = steal_pointer(&service_name);
-        n->port = HIRTE_DEFAULT_PORT;
-        LIST_HEAD_INIT(n->outstanding_requests);
-        LIST_HEAD_INIT(n->tracked_jobs);
+        Agent *agent = malloc0(sizeof(Agent));
+        if (agent != NULL) {
+                agent->ref_count = 1;
+                agent->event = steal_pointer(&event);
+                agent->user_bus_service_name = steal_pointer(&service_name);
+                agent->port = HIRTE_DEFAULT_PORT;
+        }
+
+        LIST_HEAD_INIT(agent->outstanding_requests);
```

Comment:
Not sure you should do these two lines either. I think you can just return `NULL` if `agent == NULL` and do the rest afterwards

Reply:
Yeah, no need for the indentation change if you do an early return for out-of-memory.

Reply:
Good catch!  I'll make this correction:
```
        Agent *agent = malloc0(sizeof(Agent));
        if (agent == NULL) {
                return NULL;
        }

        agent->ref_count = 1;
        agent->event = steal_pointer(&event);
        agent->user_bus_service_name = steal_pointer(&service_name);
        agent->port = HIRTE_DEFAULT_PORT;

        LIST_HEAD_INIT(agent->outstanding_requests);
        LIST_HEAD_INIT(agent->tracked_jobs);
```
And I'll make a corresponding change to manager_new() for consistency.